### PR TITLE
Hf merge master/tesseract backport max pixels 4.1.x

### PR DESCRIPTION
**Pull request by jrobble**
This change is [Reviewable](https://reviewable.io/reviews/openmpf/openmpf-components/197)

### CODE REVIEW

File: cpp/TesseractOCRTextDetection/TesseractOCRTextDetection.cpp [hunk 1]
```diff
@@ -748,20 +748,23 @@ void TesseractOCRTextDetection::rescale_image(const MPFImageJob &job, cv::Mat &i
...
 
-
-    if (max_dim * default_rescale > max_size ||
-        (ocr_fset.max_pixels > 0 &&
-        (default_rescale * im_width) * (im_height * default_rescale) > ocr_fset.max_pixels)) {
-
+    bool max_dim_exceeded = max_dim * default_rescale > max_size;
+    bool max_pix_exceeded = ocr_fset.max_pixels > 0 &&
+        (default_rescale * im_width) * (im_height * default_rescale) > ocr_fset.max_pixels;
+    if (max_dim_exceeded || max_pix_exceeded) {
```
Comment (hhuangMITRE) [inline cpp/TesseractOCRTextDetection/TesseractOCRTextDetection.cpp:754]: Looks good.

File: cpp/TesseractOCRTextDetection/TesseractOCRTextDetection.cpp [hunk 2]
```diff
@@ -774,32 +777,37 @@ void TesseractOCRTextDetection::rescale_image(const MPFImageJob &job, cv::Mat &i
...
+            string error_msg;
+            if (max_dim_exceeded) {
+                error_msg =  "[" + job.job_name + "] Unable to rescale image as one image dimension ({"
+                    + to_string(max_dim) + "}) would exceed maximum tesseract limits while the other dimension ({"
+                    + to_string(min_dim) + "}) would fall below minimum OCR-readable limits if rescaled to fit.";
+            } else if (max_pix_exceeded) {
+                error_msg =  "[" + job.job_name + "] Unable to rescale image within limit of "
+                    + to_string(ocr_fset.max_pixels) + " max pixels.";
+            }
             throw MPFDetectionException(MPF_BAD_FRAME_SIZE, error_msg);
```
Comment (hhuangMITRE) [inline cpp/TesseractOCRTextDetection/TesseractOCRTextDetection.cpp:811]: Changes reviewed and accepted for `cpp/TesseractOCRTextDetection/TesseractOCRTextDetection.cpp`. thanks!

**Review by hhuangMITRE — APPROVED** [review]
Changes for `cpp/TesseractOCRTextDetection/TesseractOCRTextDetection.cpp` reviewed and accepted. Thanks!

**Review by brosenberg42 — APPROVED** [review]
Looks good. Please remember to merge on command line.

**Pull request merged**